sentry/src/routes/campaign.rs

```diff
@@ -1,56 +1,286 @@
+
+        // Getting the campaign as it was before the update operation, should exist
+        let campaign_being_mutated = fetch_campaign(app.pool.clone(), &campaign.id).await?.ok_or(ResponseError::NotFound)?;
+
+        // modify Campaign
+        modify_campaign(&app.pool, &campaign_being_mutated, &modified_campaign, &app.redis).await.map_err(|_| ResponseError::BadRequest("Failed to update campaign".to_string()))?;
+
+        Ok(success_response(serde_json::to_string(&campaign)?))
+    }
+
+    pub async fn modify_campaign(pool: &DbPool, campaign: &Campaign, modified_campaign: &ModifyCampaign, redis: &MultiplexedConnection) -> Result<Campaign, Error> {
+        // *NOTE*: When updating campaigns make sure sum(campaigns.map(getRemaining)) <= totalDepoisted - totalspent
+        // !WARNING!: totalSpent != sum(campaign.map(c => c.spending)) therefore we must always calculate remaining funds based on total_deposit - lastApprovedNewState.spenders[user]
+        // *NOTE*: To close a campaign set campaignBudget to campaignSpent so that spendable == 0
+
+        if let Some(new_budget) = modified_campaign.budget {
+            let old_remaining = get_remaining_for_campaign(&redis, campaign.id).await?.ok_or(Error::FailedUpdate("No remaining entry for campaign".to_string()))?;
+
+            let campaign_spent = campaign.budget.checked_sub(&old_remaining).ok_or(Error::Calculation)?;
+            if campaign_spent >= new_budget {
+                return Err(Error::NewBudget("New budget should be greater than the spent amount".to_string()));
+            }
+
+            // Separate variable for clarity
+            let old_budget = campaign.budget;
+
+            match new_budget.cmp(&old_budget) {
+                Ordering::Greater | Ordering::Equal => {
+                    let new_remaining = old_remaining.checked_add(&new_budget.checked_sub(&old_budget).ok_or(Error::Calculation)?).ok_or(Error::Calculation)?;
+                    let amount_to_incr = new_remaining.checked_sub(&old_remaining).ok_or(Error::Calculation)?;
+                    increase_remaining_for_campaign(&redis, campaign.id, amount_to_incr).await?;
+                },
+                Ordering::Less => {
+                    let new_remaining = old_remaining.checked_add(&old_budget.checked_sub(&new_budget).ok_or(Error::Calculation)?).ok_or(Error::Calculation)?;
+                    let amount_to_decr = new_remaining.checked_sub(&old_remaining).ok_or(Error::Calculation)?;
+                    let decreased_remaining = decrease_remaining_for_campaign(&redis, campaign.id, amount_to_decr).await.ok_or(Error::FailedUpdate("Could't decrease remaining".to_string()))?;
+                    // If it goes below 0 it will still return 0
+                    if decreased_remaining.eq(&UnifiedNum::from_u64(0)) {
+                        return Err(Error::NewBudget("No budget remaining after decreasing".to_string()));
```

Comment:
I have to think about this error. If we have updated redis already, this will fail and it wont update the Postgres record. We must be careful here or this could cause issues down the line